updates to template docs (#743)
actualbudget/docs: docs/experimental/goal-templates.md

|`#template 500 by 2024-12 spend from 2024-03 repeat every year`| |   
|`#template 500 by 2024-12 spend from 2024-03 repeat every 2 years`| |    

### Periodic Type
If you have bills that are on a regular interval, byweekly, every 6 weeks, quarterly, once per year, this is the template for you!
This template is like the simple template but it will be based on a defined period instead of budgeted once per month.
You set a period and the start day, and every period, starting from your start day, you will get the requested amount budgeted.
Periods can be set for any number of days, weeks, months, or years.
See the table below for examples.

<!-- prettier-ignore -->
|Syntax| Budgeted Amount | Note |
|---|:---:| :---: |
|`#template 10 repeat every week starting 2025-01-06` | $ 50 | When budgeting in March 2025 |
|`#template 100 repeat every 2 months starting 2025-01-01` | $0 | When budgeting in February 2025
|`#template 100 repeat every 2 months starting 2025-01-01` | $100 | When budgeting in March 2025

As you can see, the template will budget based on defined period.

> The Periodic template also supports limits the same way the simple template does.

Reviewer: ```suggestion
The Periodic template also supports limiting, just as the simple template does.
```

Author: I think this is less clear than the original.  The name "limits" is what the template feature is referred to.

> For example if you budget in January with a limited template:

Reviewer: ```suggestion
For example, if you budget in January with a limited template:
```

### Remainder Type

> The remainder template is run differently to the other templates. Any remainder templates will be forced to run last in their own pass.

Reviewer: ```suggestion
The remainder template will run differently from the other templates. Any remainder templates will be forced to run last in their own pass.
```

> This way the amount left to budget is whatever remains after all other templates have had a chance to run.

Reviewer: ```suggestion
This way, the amount left to budget is the amount that remains after all other templates have had a chance to run.
```

Remainders will respect limits set by an `up to`.

> This limit can be set by ther remainder template, or by any other template that supports limits.

Reviewer: ```suggestion
This limit can be set by the remainder template or by any other template that supports limits.
```

> If all remainder templates are in a category with a limit, there may be funds leftover after they have run.

Reviewer: ```suggestion
If all remainder templates are in a category with a limit, there may be funds left over after they have run.
```

> To avoid having funds leftover after the remainder templates have run, always have atleast one remainder template that is not limited by an `up to`.

Reviewer: ```suggestion
To avoid having funds leftover after the remainder templates have run, always have at least one remainder template that is not limited by an `up to`.
```

Remainder templates allow an optional weight.
The weight will affect how funds are distributed among multiple remainder templates.

> The templates with a higher weight will get more than a template with a lower weight.

Reviewer: It is unclear what they will get more of - but I assume funds.

```suggestion
The templates with a higher weight will receive more funds than those with a lower weight.
```

| Savings       | `#template remainder 2` | 66.66          |
| Vacation Fund | `#template remainder`   | 33.34          |

4. Spread funds over many categories, but some have limits.

| Category        | Template line           | Amount applied |
| --------------- | ----------------------- | -------------- |
| Snack Fund      | `#template remainder 3 up to 40` | 40             |
| Vacation Fund   | `#template remainder`   | 21.66          |
| Investment Fund | `#template remainder 2` | 38.34          |

Notes on using the remainder template:

- You can use as many remainder templates as you want.

> - Remainder templates don't have a priority as they are always run last.

Reviewer: ```suggestion
- Remainder templates don't have a priority as they will always run last.
```

- If no weight is provided, the weight will be defaulted to 1.

> - The amount budget by the remainder template is calculated as: `budgeted_amount=available_funds/sum_of_weights * category_weight` unless the budgeted amount is greater than the set limit. If a category hits its limit, the excess funds will be distributed to the other remainder templates based on their weight.

Reviewer: ```suggestion
- Unless the budgeted amount is greater than the set limit, the amount budgeted by the remainder template is calculated as: `budgeted_amount=available_funds/sum_of_weights * category_weight`. If a category hits its limit, the excess funds will be distributed to the other remaining templates based on their weight.
```
